src/pd/client.rs

```diff
@@ -39,21 +36,58 @@ impl RpcClient {
                 .name_prefix(thd_name!(CLIENT_PREFIX))
                 .build(),
         );
-        for _ in 0..MAX_RETRY_TIMES {
-            match validate_endpoints(Arc::clone(&env), cfg, &security_mgr) {
-                Ok((client, members)) => {
-                    return Ok(RpcClient {
-                        cluster_id: members.get_header().get_cluster_id(),
-                        leader_client: LeaderClient::new(env, security_mgr, client, members),
-                    });
-                }
-                Err(e) => {
-                    warn!("validate PD endpoints failed"; "err" => ?e);
-                    thread::sleep(Duration::from_millis(RETRY_INTERVAL_MS));
+
+        if cfg.retry_interval.0.as_millis() == 0 {
```

Comment:
IMO, I don't think we need to cache the previous error, mostly the error messages are similar. 

and the logic I prefer likes:

```
for i in 0..retry_count {
    match validate() {
       Ok =>
       Err = > {
          if i % log_every == 0 {
               log error
          }
          // sleep 0 is fine too
          sleep(retry_interval)
       }
   }
}

Reply:
Ok.